pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -203,6 +234,40 @@ func (p *staticPolicy) RemoveContainer(s state.State, containerID string) error
 	return nil
 }
 
+func (p *staticPolicy) inheritCPUs(s state.State, pod *v1.Pod, numCPUs int) (cpuset.CPUSet, error) {
+	// Get a sorted slice of all init Container cpuSets.
+	var cpuSets []*cpuset.CPUSet
+	for _, container := range pod.Spec.InitContainers {
+		containerID, err := p.containerMap.Get(pod, &container)
+		if err != nil {
+			continue
+		}
+		if cset, ok := s.GetCPUSet(containerID); ok {
+			cpuSets = append(cpuSets, &cset)
+		}
+	}
+	sort.Slice(cpuSets, func(i, j int) bool {
+		return cpuSets[i].Size() < cpuSets[j].Size()
+	})
+
+	// Inherit up to numCPUs from the largest CPUSet in cpuSets.
```

Comment:
Can you share more about why we only try and use one `CPUSet` and don't look at all of them?

Reply:
Each init container should inherit from the previous one (if possible), so each one is an intersection of the previous, with the largest one will be the one with the most inheritable CPUs in it.

As I'm writing this though, I'm realizing that (as written) this will not work correctly for pods that have multiple app containers. Once one app container inherits some CPUs from an init container, those should be off limits to future app containers.

Should be a simple modification, but my tests don't currently cover this and they probably should.

Reply:
Fixed